### PR TITLE
Use sourceforge mirrors for boost.

### PR DESCRIPTION
**Pull request by pifon2a**
Follow the fix by @drigz.

Also format bzl.

### CODE REVIEW

**Comment by drigz** [comment]
I've updated the rules_boost change to (hopefully) fix the bzip issue too. https://github.com/nelhage/rules_boost/pull/81/files

**Comment by pifon2a** [comment]
@drigz Thank you! I will try to get rid of the boost dependency: it is used here only for... compression.

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -15,243 +15,278 @@
...
-      sha256 = "ba75b07d3fd297375a6688e9a16583eb616e7a74b3d5e8791e7a222cf36ab26e",
-      strip_prefix = "bazel_rules-98ddd7e4f7c63ea0868f08bcc228463dac2f9f12",
-      urls = [
-          "https://mirror.bazel.build/github.com/antonovvk/bazel_rules/archive/98ddd7e4f7c63ea0868f08bcc228463dac2f9f12.tar.gz",
-          "https://github.com/antonovvk/bazel_rules/archive/98ddd7e4f7c63ea0868f08bcc228463dac2f9f12.tar.gz",
-      ],
-  )
+    _maybe(
+        native.http_archive,
+        name = "com_github_nelhage_boost",
```
Comment (gaschler) [inline bazel/repositories.bzl]: rename to "_boost_build"?

File: bazel/repositories.bzl [hunk 2]
```diff
@@ -15,243 +15,278 @@
...
-      strip_prefix = "rules_boost-7289bb1d8f938fdf98078297768c122ee9e11c9e",
-      urls = [
-          "https://mirror.bazel.build/github.com/nelhage/rules_boost/archive/7289bb1d8f938fdf98078297768c122ee9e11c9e.tar.gz",
-          "https://github.com/nelhage/rules_boost/archive/7289bb1d8f938fdf98078297768c122ee9e11c9e.tar.gz",
-      ],
-  )
+    SOURCEFORGE_MIRRORS = ["phoenixnap", "newcontinuum", "cfhcable", "superb-sea2", "cytranet", "iweb", "gigenet", "ayera", "astuteinternet", "pilotfiber", "svwh"]
+    _maybe(
+        native.new_http_archive,
+        name = "boost",
```
Comment (gaschler) [inline bazel/repositories.bzl]: longer name?

File: bazel/repositories.bzl [hunk 3]
```diff
@@ -15,243 +15,278 @@
...
+        native.new_http_archive,
+        name = "net_zlib_zlib",
+        build_file = "@com_github_googlecartographer_cartographer//bazel/third_party:zlib.BUILD",
+        sha256 = "6d4d6640ca3121620995ee255945161821218752b551a1a180f4215f7d124d45",
+        strip_prefix = "zlib-cacf7f1d4e3d44d871b605da3b647f07d718623f",
+        urls = [
+            "https://mirror.bazel.build/github.com/madler/zlib/archive/cacf7f1d4e3d44d871b605da3b647f07d718623f.tar.gz",
+            "https://github.com/madler/zlib/archive/cacf7f1d4e3d44d871b605da3b647f07d718623f.tar.gz",
+        ],
+    )
```
Comment (drigz) [inline bazel/repositories.bzl]: Add this stanza to fix the bzip2 issue:

```
  _maybe(
      native.new_http_archive,
      name = "org_bzip_bzip2",
      build_file = "@com_github_nelhage_boost//:BUILD.bzip2",
      sha256 = "a2848f34fcd5d6cf47def00461fcb528a0484d8edef8208d6d2e2909dc61d9cd",
      strip_prefix = "bzip2-1.0.6",
      urls = [
          "https://mirror.bazel.build/www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz",
          "http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz",
      ],
  )
```

**Pull request merged**